docker/kubernetes-tentacle/bootstrapRunner/bootstrapRunner.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+)
+
+// The bootstrapRunner applet is designed to execute a script in a specific folder
+// and format the script's output from stdout and stderr into the following format:
+// <line number>|<RFC3339Nano date time>|<"stdout" or "stderr">|<original string>
+//
+// The applet can be executed as such:
+// bootstrapRunner <working directory> <script> <script arguments>...
+//
+// Note: all arguments given after the <script> argument are passed directly to the script as arguments.
+func main() {
+	workspacePath := os.Args[1]
+	args := os.Args[2:]
+	cmd := exec.Command("bash", args[0:]...)
+	cmd.Dir = workspacePath
+	stdOutCmdReader, _ := cmd.StdoutPipe()
+	stdErrCmdReader, _ := cmd.StderrPipe()
+
+	stdOutScanner := bufio.NewScanner(stdOutCmdReader)
+	stdErrScanner := bufio.NewScanner(stdErrCmdReader)
+
+	doneStd := make(chan bool)
+	doneErr := make(chan bool)
+
+	//stdout log file
+	so, err := os.Create(workspacePath + "/stdout.log")
+	if err != nil {
+		panic(err)
+	}
+	// close fo on exit and check for its returned error
+	defer func() {
+		if err := so.Close(); err != nil {
+			panic(err)
+		}
+	}()
+	// make a write buffer
+	stdoutLogFile := bufio.NewWriter(so)
+
+	//stderr log file
+	se, err := os.Create(workspacePath + "/stderr.log")
+	if err != nil {
+		panic(err)
+	}
+	// close fo on exit and check for its returned error
+	defer func() {
+		if err := se.Close(); err != nil {
+			panic(err)
+		}
+	}()
+	// make a write buffer
+	stderrLogFile := bufio.NewWriter(se)
+
+	go reader(stdOutScanner, stdoutLogFile, &doneStd)
+	go reader(stdErrScanner, stderrLogFile, &doneErr)
+
+	err = cmd.Start()
+	if err != nil {
+		panic(err)
+	}
+
+	// Wait for output buffering first
+	<-doneStd
+	<-doneErr
+
+	err = cmd.Wait()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "Error waiting for Cmd", err)
+		os.Exit(1)
```

Comment:
We should use the original exit code from the failed command here cc. @liam-mackie

Reply:
I'm also thinking we shouldn't add an additional log here, should we? It seems like the bootstrapper script will log errors before it returns an error code?

Reply:
I mean, the bootstrapper/script being run should manage logging it's errors before returning with a non-success exit code. This applet is only used to intercept logs and format them, not add additional logs. what do you think?

Reply:
I've updated this, we now return the exit code we got from the bootstrapper command, and also we only log if it's not an error related to the failure exit code. (io error or some other weird thing).

Reply:
Yeah, this was just a speed-hack to easily handle the error while I was hacking the script together. Some notes on my understanding:
* If it's an `ExitError` (see [here](https://pkg.go.dev/os/exec#ExitError)), we should definitely just exit with the code and not log (we already redirect stdErr, so `Stderr` in the `ExitError` would actually be `null` :shrug:)
* If it's not an `ExitError`, we should _probably_ just panic with the `Err`, since it's an unrecoverable state that we didn't expect, and we _probably_ want to have the stack. Some of the ways this could happen is being unable to write to a pipe or other OS internals. Panicking _also_ ends up running any of our `defer` code, which allows us to send telemetry if we implement it in future :)

Reply:
That sounds reasonable. I'll look at making another PR to make it panic if it's not an `ExitError` 👍